src/framework/components/sound/slot.js

```diff
@@ -0,0 +1,573 @@
+     * @property {Boolean} loop If true the slot will restart when it finishes playing
+     * @property {Boolean} overlap If true then sounds played from slot will be played independently of each other. Otherwise the slot will first stop the current sound before starting the new one.
+     * @property {Boolean} isLoaded Returns true if the asset of the slot is loaded.
+     * @property {Boolean} isPlaying  Returns true if the slot is currently playing.
+     * @property {Boolean} isPaused Returns true if the slot is currently paused.
+     * @property {Boolean} isStopped Returns true if the slot is currently stopped.
+     * @property {[pc.SoundInstance]} instances An array that contains all the {@link pc.SoundInstance}s currently being played by the slot.
+     */
+    var SoundSlot = function (component, name, options) {
+        options = options || {};
+        this._component = component;
+        this._assets = component.system.app.assets;
+        this._manager = component.system.manager;
+        this._name = name || 'Untitled';
+        this._volume = options.volume !== undefined ? pc.math.clamp(Number(options.volume) || 0, 0, 1) : 1;
+        this._pitch = options.pitch !== undefined ? Math.max(0.01, Number(options.pitch) || 0) : 1;
+        this._loop = !!(options.loop !== undefined ? options.loop : false);
+        this._duration = options.duration > 0 ? options.duration : null;
+        this._startTime = Math.max(0, Number(options.startTime) || 0);
+        this._overlap = !!(options.overlap);
+        this._autoPlay = !!(options.autoPlay);
+        this._firstNode = null;
+        this._lastNode = null;
+
+        this._asset = options.asset;
+        if (this._asset instanceof pc.Asset) {
+            this._asset = this._asset.id;
+        }
+
+        this.instances = [];
+
+        pc.events.attach(this);
+    };
+
+    SoundSlot.prototype = {
+        /**
+         * @function pc.SoundSlot#play
+         * @description Plays a sound. If {@link pc.SoundSlot#overlap} is true the new sound
+         * instance will be played independently of any other instances already playing.
+         * Otherwise existing sound instances will stop before playing the new sound.
+         * @return {pc.SoundInstance} The new sound instance
+         */
+        play: function () {
+            // stop if overlap is false
+            if (!this.overlap && (this.isPlaying || this.isPaused)) {
+                this.stop();
+            }
+
+            var instance = this._createInstance();
+            this.instances.push(instance);
+
+            // if not loaded then load first
+            // and then set sound resource on the created instance
+            if (! this.isLoaded) {
+                var onLoad = function (sound) {
+                    instance.sound = sound;
+                    if (instance._playWhenLoaded) {
+                        instance.play();
+                    }
+                }.bind(this);
```

Comment:
`this` context is not used in callback, so no need to bind it, as binding increases price of calling bound function.